src/main/scala/epic/dense/CachingLookupAndAffineTransformDense.scala

```diff
@@ -66,8 +65,8 @@ case class CachingLookupAndAffineTransformDense[FV](numOutputs: Int,
 
     def activations(fv: Array[Int]) = {
       val finalVector = DenseVector.zeros[Double](numOutputs)
-      for (i <- 0 until fv.size) {
-//        val wordPosn = fv(i) -> i
+      fv.indices.foreach { i =>
```

Comment:
i honestly would prefer `for(i <- fv.indices) {` here and elsewhere, but not enough to insist.